### PR TITLE
resolver return directly when host is ip address

### PR DESCRIPTION
**Pull request by cssivision**

### CODE REVIEW

**Comment by cssivision** [comment]
fix https://github.com/bluejekyll/trust-dns/issues/197

File: resolver/src/resolver_future.rs [hunk 1]
```diff
@@ -243,6 +243,11 @@ impl ResolverFuture {
     /// # Arguments
     /// * `host` - string hostname, if this is an invalid hostname, an error will be returned.
     pub fn lookup_ip(&self, host: &str) -> LookupIpFuture {
+        // if host is a ip address, return directly.
+        if let Some(addr) = parse_literal_ip(host) {
```
Comment (bluejekyll) [inline resolver/src/resolver_future.rs]: there's a `warn!` in the `parse_literal_ip` method in the case of errors. Given that this is really intended for host names and not just IPs, can we move that `warn!` to `read_hosts_conf`?

Comment (cssivision) [reply resolver/src/resolver_future.rs]: fixed

**Comment by bluejekyll** [comment]
Looks, great! Thanks again for the PR!

**Pull request merged**